src/telemetryClient.ts

```diff
@@ -11,6 +11,7 @@ import { Utility } from "./utility";
 const packageJSON = vscode.extensions.getExtension(Constants.ExtensionId).packageJSON;
 const extensionVersion: string = packageJSON.version;
 const aiKey: string = packageJSON.aiKey;
+const errorProperties = ["Message", "error", "detailedMessage"];

```

Comment:
It seems this way the code structure doesn't easy maintain the error message. My suggestion would be either we abstract a type for error message or We have a new api wrapper to send error. Thus the caller explicitly know how to send error.